mne_icalabel/megnet/features.py

```diff
@@ -0,0 +1,291 @@
+    fline : float, optional
+        The base frequency of the line noise to detect.
+    neighbor_width : float, optional
+        Width of the frequency neighborhood around each harmonic
+        for calculating background noise, in Hz. Default is 2.0 Hz.
+    threshold_factor : float, optional
+        Multiplicative factor for setting the detection threshold.
+        The threshold is set as the mean of the neighboring frequencies plus
+        `threshold_factor` times the standard deviation. Default is 3.
+    show : bool, optional
+        Whether to plot the PSD for channels affected by line noise.
+
+    Returns
+    -------
+    bool
+        Returns True if line noise is detected in any channel, otherwise False.
+
+    """
+    psd = raw.compute_psd(picks=picks)
+    freqs = psd.freqs
+    psds = psd.get_data()
+    ch_names = psd.ch_names
+
+    # Compute Nyquist frequency and determine maximum harmonic
+    nyquist_freq = raw.info["sfreq"] / 2.0
+    max_harmonic = int(nyquist_freq // fline)
+
+    # Generate list of harmonic frequencies based on the fundamental frequency
+    line_freqs = np.arange(fline, fline * (max_harmonic + 1), fline)
+    freq_res = freqs[1] - freqs[0]
+    n_neighbors = int(np.ceil(neighbor_width / freq_res))
+
+    line_noise_detected = []
+    for ch_idx in range(psds.shape[0]):
+        psd_ch = psds[ch_idx, :]  # PSD for the current channel
+        for lf in line_freqs:
+            # Find the frequency index closest to the current harmonic
+            idx = np.argmin(np.abs(freqs - lf))
+            # Get index range for neighboring frequencies,
+            # excluding the harmonic frequency itself
+            idx_range = np.arange(
+                max(0, idx - n_neighbors), min(len(freqs), idx + n_neighbors + 1)
+            )
+            idx_neighbors = idx_range[idx_range != idx]
+            # Calculate mean and standard deviation of neighboring frequencies
+            neighbor_mean = np.mean(psd_ch[idx_neighbors])
+            neighbor_std = np.std(psd_ch[idx_neighbors])
+
+            threshold = neighbor_mean + threshold_factor * neighbor_std
+            if psd_ch[idx] > threshold:
+                line_noise_detected.append(
+                    {"channel": ch_names[ch_idx], "frequency": lf}
+                )
+
+    if show and line_noise_detected:
+        affected_channels = set([item["channel"] for item in line_noise_detected])
+        plt.figure(figsize=(12, 3))
+        for ch_name in affected_channels:
+            ch_idx = ch_names.index(ch_name)
+            plt.semilogy(freqs, psds[ch_idx, :], label=ch_name)
+        plt.axvline(fline, color="k", linestyle="--", lw=3, alpha=0.3)
+        plt.xlabel("Frequency (Hz)")
+        plt.ylabel("Power Spectral Density (PSD)")
+        plt.legend()
+        plt.show()
+
+    return line_noise_detected
+
+
+def _check_notch(
+    raw: BaseRaw, picks: str, neighbor_width: float = 2.0, threshold_factor: float = 3
+) -> bool:
```

Comment:
I'm wondering why we don't just notch filter for them at the `line_noise` and its harmonics?

Would this impact the results?

I guess I can imagine a user constructing the Raw object via `RawArray`, which wouldn't need them to specify the line noise frequency, so maybe this wouldn't work all the time.

Reply:
Elekta/MEGIN system do store the line noise frequency, but I don't know if this is the case for other systems. I would keep it is as it is now, with the responsibility to apply the notch filter left to the user.